rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp

```diff
@@ -844,14 +843,19 @@ error_code sceNpTrophyGetTrophyInfo(u32 context, u32 handle, s32 trophyId, vm::p
 		{
 			found = true;
 
-			if (n->GetAttribute("hidden")[0] == 'y' && !ctxt->tropusr->GetTrophyUnlockState(trophyId)) // Trophy is hidden
+			const b8 hidden = n->GetAttribute("hidden")[0] == 'y';
+			const b8 unlocked = !!ctxt->tropusr->GetTrophyUnlockState(trophyId);
```

Comment:
b8 wasn't meant to be used a general bool type. it's just for ps3 object structures.
You are actually preventing the compiler to optimize boolean expresssions by using b8 instead of bool.

Reply:
in case you didn't notice, both values are passed to b8 later

Reply:
it doesnt change my point, because they are still checked in if statements.

Reply:
so 2 checks for b8 beats double GetAttribute and GetUnlockState ?

Reply:
use bool ffs.

Reply:
const bool n->GetAttribute("hidden")[0] == 'y';